pkg/kubelet/volumemanager/reconciler/reconciler.go

```diff
@@ -575,7 +575,7 @@ func (rc *reconciler) updateStates(volumesNeedUpdate map[api.UniqueVolumeName]*r
 			continue
 		}
 		if volume.pluginIsAttachable {
-			err = rc.actualStateOfWorld.MarkDeviceAsMounted(volume.volumeName)
+			err = rc.actualStateOfWorld.MarkDeviceAsMounted(volume.volumeName, volume.devicePath)
```

Comment:
No quick and clean solution here. If you use the devicePath from the API server, for RBD and Azure it will be incorrect and DeviceOpened check on unmount will fail which is what you are trying to avoid. If you set the devicePath on recreation to empty string, the DeviceOpened check will be skipped altogether on unmount for all volumes, even if we were able to retrieve the correct device path from API server.

Really if we were to fix this properly the "devicePath" passed from the controller should be a "devicePathHint" meaning it may be a devicePath or it may be a hint that can be used by the plugin on the kubelet to get the real device path. Then we can extend the Attacher interface with a `GetDevicePath(devicePathHint string)` method that every plugin implements. For most plugins this would just return the `devicePathHint`, for Azure and RDB it would return the real device path using the hint.